R/FilterStateChoices.R

```diff
@@ -171,15 +171,18 @@ ChoicesFilterState <- R6::R6Class( # nolint
       if (inherits(x, "POSIXt")) {
         private$tzone <- Find(function(x) x != "", attr(as.POSIXlt(x), "tzone"))
       }
+      private$set_is_choice_limited(x, choices)
+      x <- x[x %in% choices]
+      selected <- selected[selected %in% choices]
 
       if (!is.factor(x)) {
         x <- factor(as.character(x), levels = as.character(sort(unique(x))))
       }
       x <- droplevels(x)
-      choices <- table(x)
-      private$set_choices(names(choices))
-      self$set_selected(names(choices))
-      private$set_choices_counts(unname(choices))
+      choices_table <- table(x)
+      private$set_choices(names(choices_table))
+      self$set_selected(selected)
+      private$set_choices_counts(unname(choices_table))
```

Comment:
looks like `private$set_choices_count` is the only "setter" which suppose to be called in ChoicesFilterState